File: tutorials/07_cicd/assets/scripts/smoke-check.js
import assert from "node:assert/strict";
import { readFile } from "node:fs/promises";

const featureFlags = JSON.parse(await readFile("src/feature-flags.json", "utf8"));
const processDoc = await readFile("docs/process.md", "utf8");

assert.equal(typeof featureFlags.sampleActionEnabled, "boolean", "sampleActionEnabled flag must be boolean");
assert.match(processDoc, /1\./, "process document must contain step 1");
assert.match(processDoc, /2\./, "process document must contain step 2");
assert.match(processDoc, /3\./, "process document must contain step 3");
assert.match(processDoc, /4\./, "process document must contain step 4");

console.log("smoke check passed");

File: GitLab-Onboarding-Lectures/tutorials/CH07-CICD-Quality-Gates-and-Self-Managed-Operations/assets/scripts/smoke_check.py
from pathlib import Path


process_doc = Path("docs/process.md").read_text(encoding="utf-8")
feature_doc = Path("docs/feature-flags.md").read_text(encoding="utf-8")

if "1." not in process_doc or "4." not in process_doc:
    raise SystemExit("process steps must keep numbered structure")

if "sample_action" not in feature_doc:
    raise SystemExit("feature flag document must mention sample_action")

print("smoke check passed")
